### PR TITLE
chore: Tiered fixes

### PR DESCRIPTION
**Pull request by romange**
1. Add background offloading stats
2. remove direct_fd override - helio is already updated with default=false, so it's not needed anymore.
3. remove redundant tiered_storage_memory_margin flag

### CODE REVIEW

**Review by dranikpg — APPROVED** [review]

**Review by chakaz — COMMENTED** [review]
I see some tiering changes, disabling of top keys, and changes to dfly bench. Did you have a bad rebase?
If not, please update the title and description :rofl:

**Comment by romange** [comment]
yeah, sorry it's my wip PR

File: src/server/tiered_storage.cc [hunk 1]
```diff
@@ -590,7 +589,6 @@ PrimeValue TieredStorage::Warmup(DbIndex dbid, PrimeValue::CoolItem item) {
 
   // Bring it back to the PrimeTable.
   DCHECK(hot.ObjType() == OBJ_STRING);
-  hot.SetTouched(true);
```
Comment (dranikpg) [inline src/server/tiered_storage.cc:593]: Not sure I understand why 🤔 So we're okay with having it offloaded immediately

File: src/server/tiered_storage.cc [hunk 2]
```diff
@@ -560,7 +559,7 @@ size_t TieredStorage::ReclaimMemory(size_t goal) {
 }
 
 bool TieredStorage::ShouldStash(const PrimeValue& pv) const {
-  auto disk_stats = op_manager_->GetStats().disk_stats;
+  const auto& disk_stats = op_manager_->GetStats().disk_stats;
```
Comment (dranikpg) [inline src/server/tiered_storage.cc:562]: GetStats() returns by value so that was free 🙂

File: src/server/string_family.h [hunk 3]
```diff
@@ -36,6 +36,10 @@ struct StringValue {
   static StringValue Read(DbIndex dbid, std::string_view key, const PrimeValue& pv,
                           EngineShard* es);
 
+  bool IsFuturized() const {
+    return std::holds_alternative<util::fb2::Future<std::string>>(v_);
+  }
+
```
Comment (dranikpg) [inline src/server/string_family.h:42]: nit: not used

**Review by dranikpg — APPROVED** [review]

File: src/server/tiered_storage.cc (hunk 2 above)
Comment (romange) [reply src/server/tiered_storage.cc:562]: oh

File: src/server/tiered_storage.cc (hunk 1 above)
Comment (romange) [reply src/server/tiered_storage.cc:593]: it was a mistake, the attribute is attached to a key, not value and it's been set on a key in db_slice.cc when it's been looked up. so it's gonna be done automatically.

**Pull request merged**